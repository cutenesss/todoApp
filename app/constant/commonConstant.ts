export enum PRIORITY_LEVEL {
    HIGH = '3',
    NORMAL = '2',
    LOW = '1',
}

export enum NAME_SORT_ORDER {
    A_Z = 'A-Z',
    Z_A = 'Z-A',
}

export enum PRIORITY_SORT_ORDER {
    INCREASE = 'Increase',
    DECREASE = 'Decrease',
}