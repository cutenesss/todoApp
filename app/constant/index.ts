export * from './routerConstants'
export * from './commonConstant'
