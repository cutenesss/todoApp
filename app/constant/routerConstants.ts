export enum SCREEN_ROUTER_APP {
  ADD_EDIT_TASK_SCREEN = 'ADD_EDIT_TASK_SCREEN',
  HOME = 'HOME',
}

export const LIMIT_ITEM_PER_PAGE = 10
