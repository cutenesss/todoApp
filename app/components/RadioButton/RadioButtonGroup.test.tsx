import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react-native';
import { RadioButtonGroup } from '.';
import { PRIORITY_LEVEL } from '../../constant';

const listPriorityLevel = [
    {
        id: '1',
        label: PRIORITY_LEVEL.HIGH,
        value: PRIORITY_LEVEL.HIGH,
        selected: false
    },
    {
        id: '2',
        label: PRIORITY_LEVEL.NORMAL,
        value: PRIORITY_LEVEL.NORMAL,
        selected: false

    },
    {
        id: '3',
        label: PRIORITY_LEVEL.LOW,
        value: PRIORITY_LEVEL.LOW,
        selected: false
    }
]

describe("<RadioButtonGroup />", () => {
    const mockOnPress = jest.fn();
    const defaultProps = {
        title: 'Priority',
        initList: listPriorityLevel,
        onPress: mockOnPress,
        initValue: undefined,
        disabled: false
    }
    test('RadioButtonGroup correctly', async () => {
        const tree = renderer.create(
            <RadioButtonGroup {...defaultProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // test('onPress works correctly', () => {
    //     render(<RadioButtonGroup {...defaultProps}/>);
    //     fireEvent.press(screen.getByText("+"));
    //     expect(mockOnPress).toBeCalledTimes(1);
    // });
});