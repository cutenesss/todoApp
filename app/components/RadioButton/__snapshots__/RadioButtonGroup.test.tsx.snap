// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButtonGroup /> RadioButtonGroup correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "marginBottom": 4,
        "marginTop": 16,
      }
    }
  >
    Priority
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "column",
        },
        Object {
          "alignItems": "flex-start",
        },
      ]
    }
  />
</View>
`;
